___Execucao_com 2 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 48.279492321424186
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:48
	Energy:	0.0020 kWh
	CO2eq:	0.9481 g
	This is equivalent to:
	0.0088 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:48
	Energy:	0.0020 kWh
	CO2eq:	0.9481 g
	This is equivalent to:
	0.0088 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 2 Treads_concluida___
___Execucao_com 4 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.835572098381817
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0014 kWh
	CO2eq:	0.6593 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0014 kWh
	CO2eq:	0.6593 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 4 Treads_concluida___
___Execucao_com 6 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 32.384348170831800
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0012 kWh
	CO2eq:	0.5750 g
	This is equivalent to:
	0.0053 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0012 kWh
	CO2eq:	0.5750 g
	This is equivalent to:
	0.0053 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 6 Treads_concluida___
___Execucao_com 8 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 31.019515278749168
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0012 kWh
	CO2eq:	0.5916 g
	This is equivalent to:
	0.0055 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0012 kWh
	CO2eq:	0.5916 g
	This is equivalent to:
	0.0055 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 8 Treads_concluida___
___Execucao_com 10 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 31.505893774330616
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0012 kWh
	CO2eq:	0.5880 g
	This is equivalent to:
	0.0055 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0012 kWh
	CO2eq:	0.5880 g
	This is equivalent to:
	0.0055 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 10 Treads_concluida___
___Execucao_com 12 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 30.624409887939692
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6210 g
	This is equivalent to:
	0.0058 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6210 g
	This is equivalent to:
	0.0058 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 12 Treads_concluida___
___Execucao_com 14 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 30.236096472479403
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:30
	Energy:	0.0013 kWh
	CO2eq:	0.6219 g
	This is equivalent to:
	0.0058 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:30
	Energy:	0.0013 kWh
	CO2eq:	0.6219 g
	This is equivalent to:
	0.0058 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 14 Treads_concluida___
___Execucao_com 16 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 32.405207021161914
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6418 g
	This is equivalent to:
	0.0060 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6418 g
	This is equivalent to:
	0.0060 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 16 Treads_concluida___
___Execucao_com 18 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 31.134811892174184
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6142 g
	This is equivalent to:
	0.0057 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6142 g
	This is equivalent to:
	0.0057 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 18 Treads_concluida___
___Execucao_com 20 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 30.720309459604323
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6333 g
	This is equivalent to:
	0.0059 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6333 g
	This is equivalent to:
	0.0059 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 20 Treads_concluida___
___Execucao_com 22 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 30.925140509381890
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6301 g
	This is equivalent to:
	0.0059 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0013 kWh
	CO2eq:	0.6301 g
	This is equivalent to:
	0.0059 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 22 Treads_concluida___
___Execucao_com 24 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 31.764683469198644
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6562 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6562 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 24 Treads_concluida___
___Execucao_com 26 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 32.301729774102569
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6586 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6586 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 26 Treads_concluida___
___Execucao_com 28 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 32.159867294132710
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6602 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6602 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 28 Treads_concluida___
___Execucao_com 30 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 31.836871966719627
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6584 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:32
	Energy:	0.0014 kWh
	CO2eq:	0.6584 g
	This is equivalent to:
	0.0061 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 30 Treads_concluida___
___Execucao_com 32 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 31.410210918635130
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0014 kWh
	CO2eq:	0.6472 g
	This is equivalent to:
	0.0060 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:31
	Energy:	0.0014 kWh
	CO2eq:	0.6472 g
	This is equivalent to:
	0.0060 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 32 Treads_concluida___
___Execucao_com 34 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 32.995358400046825
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6794 g
	This is equivalent to:
	0.0063 km travelled by car
CarbonTracker: WARNING - Failed to retrieve carbon intensity: Defaulting to average carbon intensity 475 gCO2/kWh.
CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: Porto Alegre, Rio Grande do Sul, BR.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6794 g
	This is equivalent to:
	0.0063 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 34 Treads_concluida___
___Execucao_com 36 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 32.599223032593727
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6772 g
	This is equivalent to:
	0.0063 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6772 g
	This is equivalent to:
	0.0063 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 36 Treads_concluida___
___Execucao_com 38 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 33.358790657483041
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0015 kWh
	CO2eq:	0.6951 g
	This is equivalent to:
	0.0065 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0015 kWh
	CO2eq:	0.6951 g
	This is equivalent to:
	0.0065 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 38 Treads_concluida___
___Execucao_com 40 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 33.193598240613937
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6856 g
	This is equivalent to:
	0.0064 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6856 g
	This is equivalent to:
	0.0064 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 40 Treads_concluida___
___Execucao_com 42 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 32.753017138689756
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6827 g
	This is equivalent to:
	0.0064 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0014 kWh
	CO2eq:	0.6827 g
	This is equivalent to:
	0.0064 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 42 Treads_concluida___
___Execucao_com 44 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 33.489919433370233
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:34
	Energy:	0.0015 kWh
	CO2eq:	0.6981 g
	This is equivalent to:
	0.0065 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:34
	Energy:	0.0015 kWh
	CO2eq:	0.6981 g
	This is equivalent to:
	0.0065 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 44 Treads_concluida___
___Execucao_com 46 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 33.375041401013732
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0015 kWh
	CO2eq:	0.7214 g
	This is equivalent to:
	0.0067 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:33
	Energy:	0.0015 kWh
	CO2eq:	0.7214 g
	This is equivalent to:
	0.0067 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 46 Treads_concluida___
___Execucao_com 48 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 33.488551912829280
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:34
	Energy:	0.0015 kWh
	CO2eq:	0.7243 g
	This is equivalent to:
	0.0067 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:34
	Energy:	0.0015 kWh
	CO2eq:	0.7243 g
	This is equivalent to:
	0.0067 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 48 Treads_concluida___
___Execucao_com 50 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 34.275082089006901
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:34
	Energy:	0.0018 kWh
	CO2eq:	0.8387 g
	This is equivalent to:
	0.0078 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:34
	Energy:	0.0018 kWh
	CO2eq:	0.8387 g
	This is equivalent to:
	0.0078 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 50 Treads_concluida___
___Execucao_com 52 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 34.780027417466044
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0020 kWh
	CO2eq:	0.9292 g
	This is equivalent to:
	0.0086 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0020 kWh
	CO2eq:	0.9292 g
	This is equivalent to:
	0.0086 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 52 Treads_concluida___
___Execucao_com 54 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.294322005473077
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0021 kWh
	CO2eq:	0.9795 g
	This is equivalent to:
	0.0091 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0021 kWh
	CO2eq:	0.9795 g
	This is equivalent to:
	0.0091 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 54 Treads_concluida___
___Execucao_com 56 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.794564933516085
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0020 kWh
	CO2eq:	0.9595 g
	This is equivalent to:
	0.0089 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0020 kWh
	CO2eq:	0.9595 g
	This is equivalent to:
	0.0089 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 56 Treads_concluida___
___Execucao_com 58 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 34.945012724027038
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0023 kWh
	CO2eq:	1.0899 g
	This is equivalent to:
	0.0101 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0023 kWh
	CO2eq:	1.0899 g
	This is equivalent to:
	0.0101 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 58 Treads_concluida___
___Execucao_com 60 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.680998357012868
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0022 kWh
	CO2eq:	1.0348 g
	This is equivalent to:
	0.0096 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0022 kWh
	CO2eq:	1.0348 g
	This is equivalent to:
	0.0096 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 60 Treads_concluida___
___Execucao_com 62 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.129312248900533
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0022 kWh
	CO2eq:	1.0637 g
	This is equivalent to:
	0.0099 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0022 kWh
	CO2eq:	1.0637 g
	This is equivalent to:
	0.0099 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 62 Treads_concluida___
___Execucao_com 64 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 36.759858776815236
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0024 kWh
	CO2eq:	1.1528 g
	This is equivalent to:
	0.0107 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0024 kWh
	CO2eq:	1.1528 g
	This is equivalent to:
	0.0107 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 64 Treads_concluida___
___Execucao_com 66 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.601119561120868
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0022 kWh
	CO2eq:	1.0410 g
	This is equivalent to:
	0.0097 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0022 kWh
	CO2eq:	1.0410 g
	This is equivalent to:
	0.0097 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 66 Treads_concluida___
___Execucao_com 68 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.901774791069329
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0022 kWh
	CO2eq:	1.0252 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0022 kWh
	CO2eq:	1.0252 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 68 Treads_concluida___
___Execucao_com 70 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 36.246345655992627
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0023 kWh
	CO2eq:	1.0859 g
	This is equivalent to:
	0.0101 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0023 kWh
	CO2eq:	1.0859 g
	This is equivalent to:
	0.0101 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 70 Treads_concluida___
___Execucao_com 72 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 34.947912295348942
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0023 kWh
	CO2eq:	1.0807 g
	This is equivalent to:
	0.0101 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:35
	Energy:	0.0023 kWh
	CO2eq:	1.0807 g
	This is equivalent to:
	0.0101 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 72 Treads_concluida___
___Execucao_com 74 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.478017291054130
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0021 kWh
	CO2eq:	1.0200 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0021 kWh
	CO2eq:	1.0200 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 74 Treads_concluida___
___Execucao_com 76 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 36.290302615612745
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0021 kWh
	CO2eq:	1.0198 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0021 kWh
	CO2eq:	1.0198 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 76 Treads_concluida___
___Execucao_com 78 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.555917186662555
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0020 kWh
	CO2eq:	0.9470 g
	This is equivalent to:
	0.0088 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0020 kWh
	CO2eq:	0.9470 g
	This is equivalent to:
	0.0088 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 78 Treads_concluida___
___Execucao_com 80 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 36.463841564022005
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0021 kWh
	CO2eq:	1.0203 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0021 kWh
	CO2eq:	1.0203 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 80 Treads_concluida___
___Execucao_com 82 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 36.930549755692482
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0023 kWh
	CO2eq:	1.1065 g
	This is equivalent to:
	0.0103 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0023 kWh
	CO2eq:	1.1065 g
	This is equivalent to:
	0.0103 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 82 Treads_concluida___
___Execucao_com 84 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 36.672044248320162
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0022 kWh
	CO2eq:	1.0249 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0022 kWh
	CO2eq:	1.0249 g
	This is equivalent to:
	0.0095 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 84 Treads_concluida___
___Execucao_com 86 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 35.892455588094890
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0021 kWh
	CO2eq:	1.0008 g
	This is equivalent to:
	0.0093 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:36
	Energy:	0.0021 kWh
	CO2eq:	1.0008 g
	This is equivalent to:
	0.0093 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 86 Treads_concluida___
___Execucao_com 88 Treads___:
CarbonTracker: The following components were found: CPU with device(s) cpu:1, cpu:0.
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.


FFT_SERIAL
  C version

  Demonstrate an implementation of the Fast Fourier Transform
  of a complex data vector.

  Accuracy check:

    FFT ( FFT ( X(1:N) ) ) == N * X(1:N)

             N      NITS    Error

             2     10000  7.859082e-17
             4     10000  1.209837e-16
             8     10000  6.820795e-17
            16     10000  1.438671e-16
            32      1000  1.331210e-16
            64      1000  1.776545e-16
           128      1000  1.929043e-16
           256      1000  2.092319e-16
           512       100  1.927488e-16
          1024       100  2.312093e-16
          2048       100  2.445006e-16
          4096       100  2.476589e-16
          8192        10  2.574829e-16
         16384        10  2.734203e-16
         32768        10  2.918702e-16
         65536        10  2.834738e-16
        131072         1  3.143773e-16
        262144         1  3.213082e-16
        524288         1  3.283503e-16
       1048576         1  3.285515e-16
       2097152         1  3.509360e-16
       4194304         1  3.559347e-16
       8388608         1  3.710184e-16
      16777216         1  3.652753e-16
      33554432         1  3.815875e-16
      67108864         1  4.071100e-16
Execution Time: 37.099558013491333
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Average carbon intensity during training was 475.00 gCO2/kWh at detected location: UNDETECTED.
CarbonTracker: 
Actual consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0022 kWh
	CO2eq:	1.0334 g
	This is equivalent to:
	0.0096 km travelled by car
CarbonTracker: INFO - Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/carbontracker/emissions/intensity/intensity.py", line 94, in carbon_intensity
    raise exceptions.IPLocationError("Failed to retrieve location based on IP.")
carbontracker.exceptions.IPLocationError: Failed to retrieve location based on IP.

CarbonTracker: Live carbon intensity could not be fetched at detected location: Unknown. Defaulted to average carbon intensity for world in 2019 of 475.00 gCO2/kWh. at detected location: UNDETECTED.
CarbonTracker: 
Predicted consumption for 1 epoch(s):
	Time:	0:00:37
	Energy:	0.0022 kWh
	CO2eq:	1.0334 g
	This is equivalent to:
	0.0096 km travelled by car
CarbonTracker: Finished monitoring.
___Execucao_de 88 Treads_concluida___
Alerta: Menos de 10 log_bruto
